src/algoliasearch.helper.js

```diff
@@ -537,10 +537,13 @@ AlgoliaSearchHelper.prototype._getHitsSearchParams = function() {
   var additionalParams = {
     facets : facets,
     tagFilters : tagFilters,
-    distinct : this.state.distinct || false
+    distinct : this.state.distinct
   };
 
-  if( !this.state.query && facetFilters.length === 0 && numericFilters.length === 0 && tagFilters.length === 0 ) {
+  if( !this.state.query &&
```

Comment:
We might want to extract this check into a `isQueryEmpty()` method, as we're using it in several places. This will also make the intent more explicit.